app/services/district_report_service.rb

```diff
@@ -0,0 +1,83 @@
+class DistrictReportService
+  def initialize(facilities:, selected_date:)
+    @facilities = Array(facilities)
+    @selected_date = selected_date
+    @data = {
+      controlled_patients: {},
+      registrations: {},
+      cumulative_registrations: 0,
+      quarterly_registrations: []
+    }.with_indifferent_access
+  end
+
+  attr_reader :selected_date, :facilities, :data
+
+  def call
+    compile_control_and_registration_data
+
+    compile_cohort_trend_data
+
+    data
+  end
+
+  def controlled_patients(time)
+    end_range = time.end_of_month
+    mid_range = time.advance(months: -1).end_of_month
+    beg_range = time.advance(months: -2).end_of_month
+    sub_query = LatestBloodPressuresPerPatientPerMonth
+      .select("distinct on (patient_id) *")
+      .under_control
+      .order("patient_id, bp_recorded_at DESC, bp_id")
+      .where(registration_facility_id: facilities)
+      .where("(year = ? AND month = ?) OR (year = ? AND month = ?) OR (year = ? AND month = ?)",
```

Comment:
You could use `(year, month) IN (?, ?), (?, ?), (?, ?)` if you'd like. Not too big a deal either way.